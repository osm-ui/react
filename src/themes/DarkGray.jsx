import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import _ from 'lodash';
import colors from '../colors';
import { buildDarkThemeConfig } from '../helpers/themes';
import { config as defaultThemeConfig, StyledSpan } from './Default';

export const config = _.merge(
  {},
  buildDarkThemeConfig(defaultThemeConfig, colors, 'darkGray'),
  {}
);

const DarkGrayTheme = ({ children, ...rest }) => (
  <ThemeProvider theme={config}>
    <StyledSpan>
      {React.Children.map(children, child => {
        if (React.isValidElement(child)) {
          return React.cloneElement(child, rest);
        }

        return child;
      })}
    </StyledSpan>
  </ThemeProvider>
);

DarkGrayTheme.propTypes = {
  children: PropTypes.node.isRequired
};

DarkGrayTheme.displayName = 'DarkGrayTheme';

export default DarkGrayTheme;
