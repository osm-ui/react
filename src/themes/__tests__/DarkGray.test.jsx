import React from 'react';
import {
  snapshotWithElementChildren,
  snapshotWithTextChildren
} from 'helpers/tests';
import DarkGray from 'themes/DarkGray';

describe('When using snapshots', () => {
  it('Should render with an element children', () =>
    snapshotWithElementChildren(DarkGray));

  it('Should render with a text children', () =>
    snapshotWithTextChildren(DarkGray));
});
