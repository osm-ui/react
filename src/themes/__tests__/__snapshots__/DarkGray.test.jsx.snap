// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When using snapshots Should render with a text children 1`] = `
<styled.span>
  <DarkGrayTheme
    key=".0"
  >
    This is a text
  </DarkGrayTheme>
</styled.span>
`;

exports[`When using snapshots Should render with an element children 1`] = `
<styled.span>
  <DarkGrayTheme
    key=".0"
  >
    <div />
  </DarkGrayTheme>
</styled.span>
`;
